modules/context/context_response.go

```diff
@@ -136,6 +137,11 @@ func (ctx *Context) notFoundInternal(logMsg string, logErr error) {
 
 	ctx.Data["IsRepo"] = ctx.Repo.Repository != nil
 	ctx.Data["Title"] = "Page Not Found"
+
+	if git.IsErrNotExist(logErr) {
```

Comment:
Should we add `ctx.Repo != nil && ctx.Repo.Repository != nil` ?

Reply:
Coud this block of code be moved into the "repo view" handler? Put it before the "NotFound()" call.

Reply:
> Could this block of code be moved into the "repo view" handler? Put it before the "NotFound()" call.

Did some improvements in https://github.com/go-gitea/gitea/pull/26570/commits/99de298fa491b9f6423c70682914c547519dd804